src/parser.rs

```diff
@@ -415,20 +415,19 @@ impl Library {
                                         "name" => field_name = attr.value.clone(),
                                         _ => {},
                                     }
+                                    break;
                                 }
 
                                 u = Union {
                                     name: format!(
-                                        "{}{}_{}{}",
-                                        parent_name_prefix.unwrap_or(""),
-                                        record_name,
+                                        "{}_{}{}",
+                                        parent_name_prefix.unwrap_or(record_name),
                                         field_name,
                                         union_count
                                     ),
                                     c_type: Some(format!(
-                                        "{}{}_{}{}",
-                                        parent_ctype_prefix.unwrap_or(""),
-                                        c_type,
+                                        "{}_{}{}",
+                                        parent_ctype_prefix.unwrap_or(c_type),
```

Comment:
I mean the chances of there being two `GVariantBuilder` are nil, and if it has two nested unions, those are guaranteed to have different field names. If it's a struct->union->struct, then appending the fieldname + count adds uniqueness.

Unless there is a struct->union->struct->union->struct out there somewhere, I don't think there will be problems.